server/src/main/java/com/objectcomputing/checkins/services/checkin_notes/CheckinNoteServicesImpl.java

```diff
@@ -79,22 +79,23 @@ public CheckinNote read(@NotNull UUID id) {
     public CheckinNote update(@NotNull CheckinNote checkinNote) {
         MemberProfile currentUser = currentUserServices.getCurrentUser();
         boolean isAdmin = currentUserServices.isAdmin();
-        boolean isPdl = currentUserServices.hasRole(RoleType.PDL);
-
-        if (!isAdmin && !isPdl) {
-            throw new PermissionException("You do not have permission to access this resource");
-        }
 
         final UUID id = checkinNote.getId();
         final UUID checkinId = checkinNote.getCheckinid();
         final UUID createById = checkinNote.getCreatedbyid();
         CheckIn checkinRecord = checkinRepo.findById(checkinId).orElse(null);
         Boolean isCompleted = checkinRecord != null ? checkinRecord.isCompleted() : null;
         final UUID pdlId = checkinRecord != null ? checkinRecord.getPdlId() : null;
+
+        if (checkinRecord!=null && !isAdmin && !currentUser.getId().equals(checkinRecord.getTeamMemberId()) && !currentUser.getId().equals(checkinRecord.getPdlId())) {
```

Comment:
Does this require the same additional check Kimberlin mentioned on the save method?

Reply:
I think it will.